pkg/server/auth.go

```diff
@@ -63,8 +64,14 @@ func AuthUnaryInterceptor(checker AuthChecker) grpc.UnaryServerInterceptor {
 		}
 
 		var token string
-		if authHeader, ok := md["authorization"]; ok {
-			token = authHeader[0]
+		// First, let's see if the token is in our dedicated key.
+		if tokenHeader, ok := md[tokenutil.MetadataKey]; ok {
+			token = tokenHeader[0]
+		} else {
+			// Otherwise, see if it's been stuffed into Authorization.
+			if authHeader, ok := md["authorization"]; ok {
+				token = authHeader[0]
+			}
 		}
 
```

Comment:
In `AuthStreamInterceptor()` we check if `token == ""` do we need to do that here as well?

Reply:
We have some anonymous RPC endpoints that aren't streaming, that's why the non-streaming side let's it go through. The authChecker that is called here then performs another check to see if the endpoint is anonymous or not.

So basically, we have no anonymous streaming endpoints, and thus we just terminate it early.